applications/tari_base_node/src/grpc/server.rs

```diff
@@ -666,7 +778,272 @@ impl From<BlockHeader> for base_node_grpc::BlockHeader {
                 accumulated_monero_difficulty: h.pow.accumulated_monero_difficulty.into(),
                 accumulated_blake_difficulty: h.pow.accumulated_blake_difficulty.into(),
                 pow_data: h.pow.pow_data,
+                target_difficulty: h.pow.target_difficulty.as_u64(),
             }),
         }
     }
 }
+
+impl From<ChainMetadata> for base_node_grpc::MetaData {
+    fn from(meta: ChainMetadata) -> Self {
+        let diff = meta.accumulated_difficulty.unwrap_or_else(Difficulty::min);
+        Self {
+            height_of_longest_chain: meta.height_of_longest_chain.unwrap_or(0),
+            best_block: meta.best_block.unwrap_or(vec![]),
+            pruning_horizon: meta.pruning_horizon,
+            accumulated_difficulty: diff.as_u64(),
+        }
+    }
+}
+
+impl From<NewBlockTemplate> for base_node_grpc::NewBlockTemplate {
+    fn from(block: NewBlockTemplate) -> Self {
+        let header = base_node_grpc::NewBlockHeaderTemplate {
+            version: block.header.version as u32,
+            height: block.header.height,
+            prev_hash: block.header.prev_hash.clone(),
+            total_kernel_offset: Vec::from(block.header.total_kernel_offset.as_bytes()),
+            pow: Some(base_node_grpc::ProofOfWork {
+                pow_algo: match block.header.pow.pow_algo {
+                    PowAlgorithm::Monero => 0,
```

Comment:
I think you should rather implement a `From<PowAlgorithm> for u32` and a `TryFrom<u32> for PowAlgorithm` to handle these conversions. That way it is maintained in 1 clear place.

Reply:
Wel thats currently used in a from, that means we have to change the entire conversion here to a try_from. 
At this point we dont have to check account for try's.